File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
from collections import deque

def solution(progresses, speeds):
    
    result = deque()
    for i in range(len(progresses)):
        remain = 100 - progresses[i]
        day = math.ceil(remain / speeds[i])
        result.append(day)
    
    answer = []
    print(result)

    while result:
        a = result.popleft()
        count = 1
        while result and result[0] <= a:
            result.popleft()
            count += 1

        answer.append(count)
        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();

        int[] days = new int[progresses.length];
        for (int i = 0; i < progresses.length; i++) {
            int remainingProgress = 100 - progresses[i];
            double numberOfDaysRequired = (double) remainingProgress / speeds[i];
            days[i] = (int) Math.ceil(numberOfDaysRequired);
        }
        int maxDay = days[0];
        int functionCount = 1;;
        for (int i = 1; i < days.length; i++) {
            if (days[i] <= maxDay) {
                functionCount++;
                continue;
            }
            maxDay = days[i];
            answer.add(functionCount);
            functionCount = 1;
        }
        answer.add(functionCount);
        
        return answer.stream().mapToInt(i -> i).toArray();
    }
}